srtcore/core.cpp

```diff
@@ -2913,6 +2913,9 @@ void CUDT::startConnect(const sockaddr* serv_addr, int32_t forced_isn)
                 break;
             }
 
+            if (cst == CONN_REJECT)
+                sendCtrl(UMSG_SHUTDOWN);
+
```

Comment:
Should there also be a sendCtrl(UMSG_SHUTDOWN) after failing createSrtHandshake() just below?

Reply:
I have doubts about that. This that is currently handled is a predicted situation that may normally happen. The rejected state in the fail of createSrtHandshake() could only happen due to an IPE (as the log explains), and in such a situation I think no further steps should be undertaken. The connection will still break on timeout. If we had a rejection, the users will be looking for a problem in their configuration, which would be misleading for both them and us.

Reply:
OK, thanks!